finetune/base.py

```diff
@@ -194,56 +101,91 @@ def _target_model(self, *, featurizer_state, targets, n_outputs, train=False, re
+            n_examples=len(Xs) if not callable(Xs) else self.config.dataset_size,
+            batch_size=batch_size or self.config.batch_size)
+        if val_size == 0:
+            val_interval = sys.maxsize
+        val_input_fn, train_input_fn = self.input_pipeline._get_train_input_fns(Xs, Y, batch_size=batch_size,
+                                                                                val_size=val_size)
+
+        train_spec = tf.estimator.TrainSpec(input_fn=train_input_fn)
+        eval_spec = tf.estimator.EvalSpec(input_fn=val_input_fn)
+        estimator = self.get_estimator(val_interval=val_interval)
+        tf.estimator.train_and_evaluate(estimator, train_spec=train_spec,
+                                        eval_spec=eval_spec)
+
+
+    def get_estimator(self, val_interval=None, force_build_lm=False):
+        if self.estimator_ is None or self.input_pipeline.rebuild or force_build_lm:
+            hot_start = None
+            if self.input_pipeline.rebuild:
+                try:
+                    hot_start = tf.estimator.WarmStartSettings(
+                        ckpt_to_initialize_from=tf.train.latest_checkpoint(self.estimator_dir),
+                        vars_to_warm_start=[name for name, _ in tf.train.list_variables(self.estimator_dir) if
+                            not name.startswith("OptimizeLoss")])
+                    os.remove(os.path.join(self.estimator_dir, "checkpoint"))
+                except ValueError:
+                    pass
+                self.input_pipeline.rebuild = False
+
+            conf = tf.ConfigProto(
+                allow_soft_placement=self.config.soft_device_placement,
+                log_device_placement=self.config.log_device_placement,
+            )
+            num_gpus = len(self.config.visible_gpus)
+            if num_gpus > 1:
+                distribute_strategy = tf.contrib.distribute.MirroredStrategy(num_gpus_per_worker=num_gpus)
+            else:
+                distribute_strategy = None
+
+            config = tf.estimator.RunConfig(
+                tf_random_seed=self.config.seed,
+                save_summary_steps=val_interval,
+                save_checkpoints_steps=val_interval,
+                save_checkpoints_secs=None if val_interval is not None else 600,
+                session_config=conf,
+                keep_checkpoint_max=1,
+                keep_checkpoint_every_n_hours=10000,
+                log_step_count_steps=100,
+                train_distribute=distribute_strategy
+            )
+
+            print("TARGET_DIM", self.input_pipeline.target_dim)
+
+            model_fn = get_model_fn(
+                target_model_fn=self._target_model,
+                predict_op=self._predict_op,
+                predict_proba_op=self._predict_proba_op,
+                build_target_model=self.input_pipeline.target_dim is not None,
+                build_lm=force_build_lm or self.config.lm_loss_coef > 0.0 or self.input_pipeline.target_dim is None,
```

Comment:
we should store these as booleans before passing as args rather than passing in an expression as the arg